#resolution of the lidar sensor
resolution: 0.05
# topic name for lidar scan to subscribe to
topic_name: /scan/filtered
#distance to stop the robot just before the aruco  marker used for docking
dist_to_dock_station: 0.5
# slope of the marker wrt lidar used for docking
marker_slope_lidar_threshold: -165 #-120
# intensity of lidar rays when they strike the aruco marker or the docker/charger
marker_intensity: 200
# error value wrt marker intensity i.e
# marker intensity +- marker intensity threshold 
makrer_intensity_threshold: 25
# length of the aruco marker / object in use for docking which will form a clluster in the point cloud
cluster_length: 1 #0.1
# error value wrt cluster length i.e
# cluster length +- cluster_length_threshold
cluster_length_threshold: 1 #0.03
# distance from the marker of the lidar at which the robot uses lidar and not camera for docking
dist_camera_to_lidar_switch: 1
#maximum permissible translation / forward speed of the robot
speed_translation_max: 0.05
#maximum permissible angular speed of the robot
speed_angular_max: 0.1
#max permissible error in slope of the marker from the required slope of the marker in the charger.yaml file
max_delta_error_slope: 2
#width of the robot
robot_width: 0.177
#length of the robot
robot_length: 0.3
# distance of obstacle at which the robot comes to a halt
object_distance_threshold: 0.2
#minimum number of points within a given thresold distance to qualify a point as 
min_points_obstacle: 3
#max error limit outside which the revovery will start
#i.e if err_y > 0.3 -> start recovery which is to go to the docking position
max_err_y: 0.3
#kp parameter
kp: 0.05
#kd parameter
kd: 0.05
#time to wait for obstacle clearance (s)	before aborting the task
time_for_obstacle_clearance: 10

